core/src/main/java/org/infinispan/marshall/core/internal/ExternalJBossMarshaller.java

```diff
@@ -0,0 +1,226 @@
+package org.infinispan.marshall.core.internal;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.io.OutputStream;
+
+import org.infinispan.commons.io.ByteBuffer;
+import org.infinispan.commons.marshall.BufferSizePredictor;
+import org.infinispan.commons.marshall.Externalizer;
+import org.infinispan.commons.marshall.MarshallableTypeHints;
+import org.infinispan.commons.marshall.StreamingMarshaller;
+import org.infinispan.configuration.global.GlobalConfiguration;
+import org.infinispan.marshall.core.JBossMarshaller;
+import org.jboss.marshalling.ByteInput;
+import org.jboss.marshalling.ByteOutput;
+import org.jboss.marshalling.ObjectTable;
+import org.jboss.marshalling.Unmarshaller;
+
+final class ExternalJBossMarshaller implements StreamingMarshaller {
+
+   final MarshallableTypeHints marshallableTypeHints = new MarshallableTypeHints();
+   final JBossMarshaller marshaller;
```

Comment:
As discussed in the chat, I spotted ``` org.hibernate.search.query.engine.impl.LuceneHSQuery``` as one of the offending classes, being ```Serializable``` but holding internally lucene queries which we do provide externalizers. So one way of solving it for now would be to provide externalizer for those classes as well. 
Going forward, I'd not like to remove ClusteredQuery but send a string representation of the query (IQL?) over the wire to broadcast it.